addons/binding/org.openhab.binding.minecraft/src/main/java/org/openhab/binding/minecraft/discovery/MinecraftDiscoveryService.java

```diff
@@ -0,0 +1,210 @@
+/**
+ * Copyright (c) 2014-2016 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.minecraft.discovery;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.eclipse.smarthome.config.discovery.AbstractDiscoveryService;
+import org.eclipse.smarthome.config.discovery.DiscoveryResult;
+import org.eclipse.smarthome.config.discovery.DiscoveryResultBuilder;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.openhab.binding.minecraft.MinecraftBindingConstants;
+import org.openhab.binding.minecraft.config.ServerConfig;
+import org.openhab.binding.minecraft.handler.MinecraftServerHandler;
+import org.openhab.binding.minecraft.handler.server.ServerConnection;
+import org.openhab.binding.minecraft.internal.MinecraftHandlerFactory;
+import org.openhab.binding.minecraft.message.data.PlayerData;
+import org.openhab.binding.minecraft.message.data.SignData;
+import org.openhab.binding.minecraft.util.Pair;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import rx.Observable;
+import rx.Subscription;
+import rx.functions.Action1;
+import rx.functions.Func1;
+import rx.functions.Func2;
+import rx.schedulers.Schedulers;
+import rx.subscriptions.CompositeSubscription;
+
+/**
+ * Handles discovery of Minecraft server, players and signs.
+ *
+ * @author Mattias Markehed
+ */
+public class MinecraftDiscoveryService extends AbstractDiscoveryService {
+
+    private final Logger logger = LoggerFactory.getLogger(MinecraftDiscoveryService.class);
+
+    private static final int DISCOVER_TIMEOUT_SECONDS = 60;
+
+    private CompositeSubscription subscription;
+
+    public MinecraftDiscoveryService() {
+        super(MinecraftBindingConstants.SUPPORTED_THING_TYPES_UIDS, DISCOVER_TIMEOUT_SECONDS, false);
```

Comment:
This isn't correct - SUPPORTED_THING_TYPES_UIDS is a singleton with the Server, but you support players and signs here.